### PR TITLE
Adding cryo fields to coupler budget accounting

### PR DESCRIPTION
**Pull request by darincomeau**
Adds five new o2x coupling fields and terms to coupler budget for polar configurations:

`Foxo_ismw` - water from ice shelf basal melting (either prognostic or data)
`Foxo_rrofl` - water removed from Antarctica liquid runoff
`Foxo_rrofi` - water removed from Antarctica solid runoff

`Foxo_ismh` - heat from ice shelf basal melting
`Foxo_rrofih` - heat from removed ice runoff

`Foxo_frazil_li` - frazil ice formed in ice-shelf cavities ('land ice frazil')
`Foxo_frazil_q` - latent heat associated with land ice frazil

These new fields are used to calculate `wpolar` and `hpolar`, which are Antarctic Ice Sheet imbalances in polar configurations. These coupler budget terms replace the existing iceberg `wberg` and `hberg` entries in the budget table, which are included in the `wpolar` and `hpolar` totals.

Adds a driver namelist option (`flds_polar`) to only send these coupling fields in polar configurations.

Changes to ocean conservation analysis member:
 - Turns on terms in ocean conservation analysis member when data ice-shelf melt fluxes are used.
 - Removes `landIceFreshwaterFluxesOn` conditional on writing land ice frazil terms
 - Removes accumulated land ice frazil salinity terms for salt conservation, since land ice frazil is assumed to have 0 salinity.

Further discussion in https://github.com/E3SM-Ocean-Discussion/E3SM/pull/74

[BFB]
[NML]

### CODE REVIEW

**Comment by darincomeau** [comment]
Here is the coupler budget table after one month of a `ne30pg2_ECwISC30to60E2r1.CRYO1850-DISMF` test:

```
(seq_diag_print_mct) NET HEAT BUDGET (W/m2): period =  monthly: date =     10201     0
                           atm            lnd            rof            ocn         ice nh         ice sh            glc        *SUM*
         hfreeze     0.00000000     0.00000000     0.00000000     0.06212584    -0.06212584     0.00000000     0.00000000     0.00000000
           hmelt     0.00000000     0.00000000     0.00000000    -0.41618844     0.33142624     0.08507855     0.00000000     0.00031635
          hnetsw  -172.11170145    35.69859350     0.00000000   136.29059592     0.06204593     0.06810322     0.00000000     0.00763712
           hlwdn  -332.39451972    80.08749536     0.00000000   247.09087671     5.01251824     0.20517240     0.00000000     0.00154300
           hlwup   388.84155761   -98.86623025     0.00000000  -283.46553532    -6.27045094    -0.23999653     0.00000000    -0.00065543
         hlatvap    83.00674983    -6.87900876     0.00000000   -76.09764783    -0.02550562    -0.00479259     0.00000000    -0.00020497
         hlatfus     0.67328674    -0.35937298     0.00000000    -0.24544797    -0.06723218    -0.00120155     0.00000000     0.00003206
          hiroff     0.00000000     0.04587322     0.00000000    -0.04559532     0.00000000     0.00000000     0.00000000     0.00027791
            hsen    19.75339851    -8.47541141     0.00000000   -11.26688638    -0.01226106     0.00067890     0.00000000    -0.00048143
           hberg     0.00000000     0.00000000     0.00000000    -0.06320892     0.00000000     0.00000000     0.00000000    -0.06320892
            hism     0.00000000     0.00000000     0.00000000     0.01125439     0.00000000     0.00000000     0.00000000     0.01125439
           hriof     0.00000000     0.00000000     0.00000000     0.03160610     0.00000000     0.00000000     0.00000000     0.03160610
        hh2otemp     0.42776435     0.00000000     0.00000000    -0.42799595     0.00000000     0.00000000     0.00000000    -0.00023160
           *SUM*   -11.80346412     1.25193868     0.00000000    11.45795284    -1.03158523     0.11304241     0.00000000    -0.01211542

(seq_diag_print_mct) NET WATER BUDGET (kg/m2s*1e6): period =  monthly: date =     10201     0
                           atm            lnd            rof            ocn         ice nh         ice sh            glc        *SUM*
         wfreeze     0.00000000     0.00000000     0.00000000    -0.18617274     0.18617274     0.00000000     0.00000000    -0.00000000
           wmelt     0.00000000     0.00000000     0.00000000    -2.45519385     2.78357026    -0.32692927     0.00000000     0.00144714
           wrain   -31.38352297     5.15651606     0.00000000    26.20419654     0.01291856     0.00159838     0.00000000    -0.00829343
           wsnow    -2.01764082     1.07693431     0.00000000     0.73553482     0.20147491     0.00360070     0.00000000    -0.00009608
           wevap    33.18177647    -2.74422912     0.00000000   -30.42688838    -0.00902837    -0.00171256     0.00000000    -0.00008195
         wrunoff     0.00000000    -0.36266438     0.17583865     0.18515025     0.00000000     0.00000000     0.00000000    -0.00167548
         wfrzrof     0.00000000    -0.13746845     0.00000000     0.13663565     0.00000000     0.00000000     0.00000000    -0.00083280
           wberg     0.00000000     0.00000000     0.00000000     0.18475440     0.00000000     0.00000000     0.00000000     0.18475440
            wism     0.00000000     0.00000000     0.00000000     0.03372607     0.00000000     0.00000000     0.00000000     0.03372607
           wrrof     0.00000000     0.00000000     0.00000000    -0.05490293     0.00000000     0.00000000     0.00000000    -0.05490293
           wriof     0.00000000     0.00000000     0.00000000    -0.09471410     0.00000000     0.00000000     0.00000000    -0.09471410
          wirrig     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000
           *SUM*    -0.21938732     2.98908842     0.17583865    -5.73787427     3.17510811    -0.32344276     0.00000000     0.05933083
```

For comparison, here are the heat and water fluxes from the ocean's conservation analysis member:

```
 HEAT FLUXES: explicit
 MPAS-Ocean name            W (raw sum)    coupler name    short name         W/m^2 (flux/A)
 frazilMass * LHF /dt       3.17316703E+13 o2x_Fioo_q      hfreeze               0.06220662
 seaIceHeatFlux            -2.12329699E+14 x2o_Fioi_melth  hmelt                -0.41625019
 shortWaveHeatFlux          6.95162481E+16 x2o_Foxx_swnet  hnetsw              136.27934277
 longWaveHeatFluxDown       1.26041452E+17 x2o_Faxa_lwdn   hlwdn               247.09109933
 longWaveHeatFluxUp        -1.44596219E+17 x2o_Foxx_lwup   hlwup              -283.46578204
 latentHeatFlux            -3.88177259E+16 x2o_Foxx_lat    hlatvap             -76.09809665
 snowFlux * LHF            -1.25177939E+14                 hlatfus              -0.24539827
 iceRunoffFlux * LHF       -7.12639389E+12                 hiroff               -0.01397055
 removedIceRunoffFlux *LHF -1.61474699E+13                 hiroff                               -0.03165543
       SUM: ice runoff*LHF -2.32738638E+13                 hiroff SUM                           -0.04562598
 sensibleHeatFlux          -5.74675264E+15 x2o_Foxx_sen    hsen                -11.26590824
 icebergHeatFlux           -3.22957603E+13 x2o_Fioi_bergh  hberg                -0.06331246
 landIceHeatFlux            5.74473761E+12                                       0.01126196
    landIceFrazilHeatFlux   0.00000000E+00 (already in hfreeze, do not sum                )      0.00000000
 SUM EXPLICIT HEAT FLUXES   6.05754912E+15                                      11.87519228

 HEAT FLUXES: implicit through temperature contributions of mass fluxes
 MPAS-Ocean name            W (sum*rho*cp) coupler name    short name        W/m^2 (flux/A)
 RainTemperatureFlux        1.17136764E+15                                       2.29634389
 EvapTemperatureFlux       -1.40340882E+15                                      -2.75123638
 SeaIceTemperatureFlux      8.80789754E+12                                       0.01726696
 RiverRunoffTempFlux        5.37659625E+12                                       0.01054026
 IcebergTemperatureFlux    -7.13326099E+11                                      -0.00139840
 SUM IMPLICIT HEAT FLUXES  -2.18570011E+14                 hh20temp             -0.42848368
 SUM IMP+EXP  HEAT FLUXES   5.83897911E+15                                      11.44670860
 
 MASS FLUXES
 MPAS-Ocean name          kg/s (F)       coupler name     short name     kg/m^2/s*1e6 (F/A)
 frazilFreshwaterFlux    -9.51422423E+07 o2x_Fioo_frazil  wfreeze             -0.18651643
 seaIceFreshwaterFlux    -1.25295544E+09 x2o_Fioi_meltw   wmelt               -2.45628825
 rainFlux                 1.33675229E+10 x2o_Faxa_rain    wrain               26.20563223
 snowFlux                 3.75121184E+08 x2o_Faxa_snow    wsnow                0.73538590
 icebergFreshwaterFlux    9.43978215E+07 x2o_Fioi_bergw   wberg                0.18505707
 evaporationFlux         -1.55208820E+10 x2o_Foxx_evap    wevap              -30.42706783
 riverRunoffFlux          6.64381039E+07 x2o_Foxx_rofl    wrunoff              0.13024496
 removedRiverRunoffFlux   2.80707623E+07                  wrunoff                              0.05502980
       SUM: river runoff  9.45088662E+07 x2o_Foxx_rofl    wrunoff SUM                          0.18527476
 iceRunoffFlux            2.13556904E+07 x2o_Foxx_rofi    wfrzrof              0.04186560
 removedIceRunoffFlux     4.83891815E+07                  wfrzrof                              0.09486194
       SUM: ice runoff    6.97448720E+07 x2o_Foxx_rofi    wfrzrof SUM                          0.13672754
 landIceFreshwaterFlux    1.72152760E+07                                       0.03374875
   landIceFrazilFlux      0.00000000E+00 (already in wfreeze, do not sum                )      0.00000000
 SUM VOLUME FLUXES       -2.92692872E+09                                      -5.73793801
 ```
 
 So as an example, coupler values: 
 `hism       0.01125439`
 `wism       0.03372607`
 
 corresponding ocean values:
 `landIceHeatFlux      0.01126196`
 `andIceFreshwaterFlux  0.03374875`

**Comment by darincomeau** [comment]
I was trying to confirm that this did not change coupler budget values in `WCYCL1850` cases when I came across https://github.com/E3SM-Project/E3SM/issues/6230. I did confirm this in the development branch, but I'll confirm again here once that issue is resolved.

File: components/mpas-ocean/src/analysis_members/Registry_conservation_check.xml
Comment (darincomeau) [reply components/mpas-ocean/src/analysis_members/Registry_conservation_check.xml:132]: here's my comment from the Ocean Discussion PR for more context:

>@mark-petersen can you check the proposed unit changes in the conservation check analysis member's Registry here https://github.com/E3SM-Ocean-Discussion/E3SM/commit/285be919629f04d40cf3402bf6e69c865d47d580?

@cbegeman and I looked into this and see the flux variables being multiplied by m^2 here:
https://github.com/E3SM-Project/E3SM/blob/50485d0c929a873b5935ea3c2dc1da4d236319dd/components/mpas-ocean/src/analysis_members/mpas_ocn_conservation_check.F#L954
and not divided out again. Also when we look at the values that are written out to the conservation check analysis member, we can back out values in kg m^-2 s^-1 that appear in the ocn.log file from this analysis member by scaling by 1/area again.

The reason we're looking here is we want to use these accumulated*Flux quantities to produce a net AIS runoff time series plot for polar simulations to compare to global ssh anomaly, and to us everything is consistent if these quantities are actually in units kg s^-1.

Also, many thanks @mark-petersen for your work on this analysis member in the past. It's made verification of this coupler budget accounting much easier.

**Comment by darincomeau** [comment]
@jonbob and I are in discussions on how to avoid sending additional coupling fields in non-polar configurations.

**Comment by darincomeau** [comment]
Added Do not merge tag as current implementation is not backwards compatible due to new drv namelist option.

**Comment by darincomeau** [comment]
Note I moved the Registry conservation check unit changes to it's own PR, since it's strictly not needed here: https://github.com/E3SM-Project/E3SM/pull/6257

**Comment by darincomeau** [comment]
## Testing, water budget

Results from a one month `ne30pg2_r05_IcoswISC30E3r5.CRYO1850-DISMF` test:

Coupler budget:
```
(seq_diag_print_mct) NET WATER BUDGET (kg/m2s*1e6): period =  monthly: date =     10201     0
                           atm            lnd            rof            ocn         ice nh         ice sh            glc        *SUM*
         wfreeze     0.00000000     0.00000000     0.00000000    -0.17216493     0.16956143     0.00260350     0.00000000     0.00000000
           wmelt     0.00000000     0.00000000     0.00000000     7.41613492     2.09865714    -9.51875614     0.00000000    -0.00396408
           wrain   -31.92914589     5.22087715     0.00000000    26.55415491     0.01481922     0.13363624     0.00000000    -0.00565837
           wsnow    -2.47057796     1.17303894     0.00000000     0.81206795     0.20403288     0.28064687     0.00000000    -0.00079133
          wpolar     0.00000000     0.00000000     0.00000000     0.14243477     0.00000000     0.00000000     0.00000000     0.14243477
           wevap    34.30774137    -2.81504210     0.00000000   -31.41785396    -0.00832346    -0.06661326     0.00000000    -0.00009142
         wrunoff     0.00000000    -0.33307737     0.18564378     0.14608528     0.00000000     0.00000000     0.00000000    -0.00134831
         wfrzrof     0.00000000    -0.13723993     0.00000000     0.13630950     0.00000000     0.00000000     0.00000000    -0.00093043
          wirrig     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000     0.00000000
           *SUM*    -0.09198248     3.10855669     0.18564378     3.61716844     2.47874721    -9.16848280     0.00000000     0.12965084
 ```
           
Ocean conservation analysis:

```
 MASS CONSERVATION CHECK

 MASS FLUXES
 MPAS-Ocean name          kg/s (F)       coupler name     short name     kg/m^2/s*1e6 (F/A)
 frazilFreshwaterFlux    -8.80103671E+07 o2x_Fioo_frazil  wfreeze             -0.17253513
 seaIceFreshwaterFlux     3.78410865E+09 x2o_Fioi_meltw   wmelt                7.41834971
 rainFlux                 1.35461045E+10 x2o_Faxa_rain    wrain               26.55572294
 snowFlux                 4.14274325E+08 x2o_Faxa_snow    wsnow                0.81214154
 icebergFreshwaterFlux    9.56158749E+07 x2o_Fioi_bergw   wberg                0.18744493
 evaporationFlux         -1.60262376E+10 x2o_Foxx_evap    wevap              -31.41776480
 riverRunoffFlux          7.38715228E+07 x2o_Foxx_rofl    wrunoff              0.14481740
 removedRiverRunoffFlux   6.96646201E+05                  wrunoff                              0.00136570
       SUM: river runoff  7.45681690E+07 x2o_Foxx_rofl    wrunoff SUM                          0.14618311
 iceRunoffFlux            2.71198684E+07 x2o_Foxx_rofi    wfrzrof              0.05316567
 removedIceRunoffFlux     4.24583264E+07                  wfrzrof                              0.08323511
       SUM: ice runoff    6.95781948E+07 x2o_Foxx_rofi    wfrzrof SUM                          0.13640078
 landIceFreshwaterFlux    2.01565440E+07                                       0.03951480
   landIceFrazilFlux      0.00000000E+00 (already in wfreeze, do not sum                )      0.00000000
 SUM VOLUME FLUXES        1.84700332E+09                                       3.62085707
 ```

**Comment by darincomeau** [comment]
The coupler `wpolar` term is 0.14243477. The freshwater AIS imbalance is the difference of the added terms, iceberg and ice-shelf melting, and the removed terms, liquid and solid runoff around Antarctica:

From the ocean conservation analysis:
`icebergFreshwaterFlux` + `landIceFreshwaterFlux` - (`removedRiverRunoffFlux` + `removedIceRunoffFlux`) = 
0.18744493 + 0.03951480 - (0.00136570 + 0.08323511) = 0.14235892

So the totals calculated by the coupler and ocean agree to the first 3 digits.

**Comment by darincomeau** [comment]
## Testing, heat budget

Results from a one month `ne30pg2_r05_IcoswISC30E3r5.CRYO1850-DISMF` test:

Coupler budget:

```
(seq_diag_print_mct) NET HEAT BUDGET (W/m2): period =  monthly: date =     10201     0
                           atm            lnd            rof            ocn         ice nh         ice sh            glc        *SUM*
         hfreeze     0.00000000     0.00000000     0.00000000     0.05745144    -0.05658265    -0.00086879     0.00000000     0.00000000
           hmelt     0.00000000     0.00000000     0.00000000    -2.49129361     0.31424845     2.17867086     0.00000000     0.00162571
          hnetsw  -167.98058150    35.33377172     0.00000000   130.72201769     0.06276344     1.87517082     0.00000000     0.01314217
           hlwdn  -333.05889584    80.25069457     0.00000000   241.20007430     5.16442560     6.45111864     0.00000000     0.00741728
           hlwup   388.02110159   -98.67948895     0.00000000  -275.95965619    -6.33974221    -7.04051362     0.00000000     0.00170063
         hlatvap    85.84263724    -7.05720630     0.00000000   -78.57605276    -0.02350521    -0.18610161     0.00000000    -0.00022865
         hlatfus     0.82443186    -0.39144309     0.00000000    -0.27098707    -0.06808577    -0.09365186     0.00000000     0.00026407
          hiroff     0.00000000     0.04579696     0.00000000    -0.04548648     0.00000000     0.00000000     0.00000000     0.00031048
            hsen    19.83278120    -8.30456890     0.00000000   -11.58711974     0.01583591     0.04259911     0.00000000    -0.00047242
          hpolar     0.00000000     0.00000000     0.00000000    -0.04956268     0.00000000     0.00000000     0.00000000    -0.04956268
        hh2otemp     0.54241775     0.00000000     0.00000000    -0.54290313     0.00000000     0.00000000     0.00000000    -0.00048537
           *SUM*    -5.97610769     1.19755602     0.00000000     2.45648176    -0.93064243     3.22642355     0.00000000    -0.02628879
```

Ocean conservation analysis:

```
 ENERGY CONSERVATION CHECK

 HEAT FLUXES: explicit
 MPAS-Ocean name            W (raw sum)    coupler name    short name         W/m^2 (flux/A)
 frazilMass * LHF /dt       2.93438460E+13 o2x_Fioo_q      hfreeze               0.05752554
 seaIceHeatFlux            -1.27125623E+15 x2o_Fioi_melth  hmelt                -2.49216504
 shortWaveHeatFlux          6.66754894E+16 x2o_Foxx_swnet  hnetsw              130.71033196
 longWaveHeatFluxDown       1.23036966E+17 x2o_Faxa_lwdn   hlwdn               241.20111959
 longWaveHeatFluxUp        -1.40767605E+17 x2o_Foxx_lwup   hlwup              -275.96018491
 latentHeatFlux            -4.00816204E+16 x2o_Foxx_lat    hlatvap             -78.57582975
 snowFlux * LHF            -1.38243342E+14                 hlatfus              -0.27101163
 iceRunoffFlux * LHF       -9.04990008E+12                 hiroff               -0.01774138
 removedIceRunoffFlux *LHF -1.41683435E+13                 hiroff                               -0.02777556
       SUM: ice runoff*LHF -2.32182436E+13                 hiroff SUM                           -0.04551694
 sensibleHeatFlux          -5.91033619E+15 x2o_Foxx_sen    hsen                -11.58659670
 icebergHeatFlux           -3.27124856E+13 x2o_Fioi_bergh  hberg                -0.06412941
 landIceHeatFlux            6.72623874E+12                                       0.01318609
    landIceFrazilHeatFlux   0.00000000E+00 (already in hfreeze, do not sum                )      0.00000000
 SUM EXPLICIT HEAT FLUXES   1.53770211E+15                                       3.01450435
 ```

**Comment by darincomeau** [comment]
The coupler `hpolar` term is -0.04956268. The AIS heat imbalance is the difference of the added terms, associated iceberg and ice-shelf melting, and the removed term, associated solid runoff around Antarctica. There is no heat flux associated with liquid runoff sent from the coupler, so there is no liquid runoff term in this budget.

From the ocean conservation analysis:
`icebergHeatFlux` + `landIceHeatFlux` - `removedIceRunoffFlux *LHF` = 
-0.06412941 - 0.01318609 - (-0.02777556) = -0.04953994

So the totals calculated by the coupler and ocean agree to the first 3 digits.

Note the sign on `landIceHeatFlux`; in the implementation here I have assumed it should be reversed. Looking in `timeSeriesStatsMonthly` I see if taking on both positive and negative values, but I would think the total sum should be negative, extracting heat from the ocean.

**Comment by darincomeau** [comment]
## Testing, no impact on non-polar configurations

Confirmed with one month `ne30pg2_r05_IcoswISC30E3r5.WCYCL1850` tests before this PR and with this PR, and values in the coupler budget do not change. The only change is `wberg`->`wpolar` and `hberg` -> `hpolar`.

**Comment by darincomeau** [comment]
## Testing, backwards compatibility 

I ran a `ne30pg2_r05_IcoswISC30E3r5.WCYCL1850` before this PR, then created a new hybrid case off this PR using restarts from the case before this PR, and the job successfully ran.
Part of the reason the `flds_polar` driver namelist was added was to prevent the model from looking for the new coupling fields in restarts that did not have them, which throws numerous errors to e3sm.log file about setting the missing coupling fields to 0 for backwards compatibility. So this avoids errors in using restarts before this PR in non-polar configurations.

In the situation where a user has an existing case before this PR, then updates the code that includes this PR, rebuilds, and tries to run, the model will error and abort looking for the new driver namelist. Adding `flds_polar = .false.` to `user_nl_cpl` will get around this.

**Review by cbegeman — APPROVED** [review]
Now that we've resolved the question of the land ice heat flux sign, I'm happy with these changes. You've shown that the coupler accounting agrees with the ocean component's accounting, and that's the main thing I was looking for. Let me know if you would like any of my help with further testing. Thanks for your work on this @darincomeau!

**Review by xylar — APPROVED** [review]
This looks good to me.  It has already caught one mistake at least (the sign of the DISMF heat flux) and will be a big improvement in Polar runs going forward.

**Comment by darincomeau** [comment]
## Additional Testing

Passes

SMS_Lm1_P512.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel
SMS_Lm1_P512.ne30pg2_r05_IcoswISC30E3r5.CRYO1850.chrysalis_intel
SMS_Lm1_P512.ne30pg2_r05_IcoswISC30E3r5.CRYO1850-DISMF.chrysalis_intel

ERS_Ld31_P512.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel
ERS_Ld31_P512.ne30pg2_r05_IcoswISC30E3r5.CRYO1850.chrysalis_intel

Also ran some G-case tests, where the budgets aren't useful but wanted to make sure nothing unexpected happened:

SMS_Lm1_P512.TL319_IcoswISC30E3r5.GMPAS-JRA1p5.chrysalis_intel
SMS_Lm1_P512.TL319_IcoswISC30E3r5.GMPAS-JRA1p5-DIB-PISMF.chrysalis_intel
SMS_Lm1_P512.TL319_IcoswISC30E3r5.GMPAS-JRA1p5-DIB-DISMF.chrysalis_intel

**Comment by darincomeau** [comment]
Thanks @cbegeman and @xylar for the review. I've updated the tables in the water and heat budget testing sections to reflect the DISMF file with the corrected latent heat flux sign.

@stephenprice @proteanplanet this is ready for review. I don't think any additional testing on your part is needed (though you're of course welcome to), visual inspection should be enough.

I'll rebase once the v3 tag is made and rerun the standard tests.

**Comment by darincomeau** [comment]
Rebased on v3.0.0

Reran and PASSes

SMS_Lm1_P512.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel
SMS_Lm1_P512.ne30pg2_r05_IcoswISC30E3r5.CRYO1850-DISMF.chrysalis_intel
SMS_Lm1_P512.ne30pg2_r05_IcoswISC30E3r5.CRYO1850.chrysalis_intel

ERS_Ld31_P512.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel
ERS_Ld31_P512.ne30pg2_r05_IcoswISC30E3r5.CRYO1850.chrysalis_intel

**Comment by cbegeman** [comment]
@darincomeau What do you think about including these changes to `driver/ocn_comp_mct.F` https://github.com/E3SM-Project/E3SM/commit/34699cdd8285a870892f48778254514efd7a5783? I think this is consistent with what we discussed: sending both sea ice and land ice frazil mass fluxes whenever non-zero. I added another conditional to avoid computations of melt potential in ice shelf cavities since that variable should only be used by the sea ice model. This commit is not yet tested.

I'll let you make the changes to `wpolar` and `hpolar` in `driver-mct/main/seq_diag_mct.F90` separately.

What do you think about including these 2 changes in this PR and I'll open a separate one to fix up the ocean analysis members (conservation check and global stats)?

**Comment by darincomeau** [comment]
@cbegeman @xylar this has been updated to include accounting for land ice frazil.

**Comment by darincomeau** [comment]
@stephenprice @proteanplanet this one is ready to merged soon, I don't think we need anymore careful reviews but wanted to give you a heads up in case you had any concerns.

@rljacob this one adds coupling fields, and a new driver namelist option. The namelist option was added to only use the new coupling fields in polar configurations, in an attempt to minimize disruption with backwards compatibility and avoid throwing errors. I also made the same changes in driver-mct to driver-moab to keep the code in sync, but I don't know how to test that - I can remove those changes if that's preferred. Also wanted to give you a heads up here in case you had any concerns.

**Comment by proteanplanet** [comment]
@darincomeau The cryo configuration is being added to https://github.com/E3SM-Project/SimulationScripts/blob/master/archive/PolarGroup/E3SM-Polar-Developer.sh, and we're also setting up an e3sm_cryo_developer test suite in https://github.com/E3SM-Project/E3SM/blob/master/cime_config/tests.py.  I didn't see any PEM or PET tests. Even though the code suggests these won't be affected, I'll run them to be sure.

**Comment by darincomeau** [comment]
Ok thanks @proteanplanet ! I agree it would be good to check PEM/PET, I've been trying to be as careful as possible here to avoid being disruptive. I can also run those since I'm already setup locally.

**Comment by darincomeau** [comment]
Actually I'm just remembering I did run the `e3sm_ice_developer` test suite at some point here which includes PEM/PET, but I didn't post results (PASSed) since it was D-cases where the coupler budgets aren't relevant and there was no ocean conservation analysis member to check against.
@jonbob I can run an integration test suite if that'd be helpful.

**Comment by jonbob** [comment]
passes:
* PET_Ln9_PS.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel.allactive-mach-pet
* ERS_Ld3.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel.allactive-nlmaps

waiting on:
* PEM_Ln9.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel

**Comment by jonbob** [comment]
Passed PEM_Ln9.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel as well. Merged to next

File: driver-mct/cime_config/config_component_e3sm.xml
Comment (darincomeau) [inline driver-mct/cime_config/config_component_e3sm.xml:182]: this logic is missing the case when DIB is not active, but DISMF or PISMF is active.

**Comment by proteanplanet** [comment]
This appears to have already been merged to next.   Can @darincomeau 's changes be included in that?

File: driver-mct/cime_config/config_component_e3sm.xml
Comment (jonbob) [reply driver-mct/cime_config/config_component_e3sm.xml:182]: Can you make an E3SM issue? I already merged this to master and have been in the process of blessing the expected NML and regular DIFFs (which are just due to a change in the field list)

**Comment by jonbob** [comment]
merged to master and expected NML and regular DIFFs blessed -- except for pm-cpu which may not report today

File: driver-mct/cime_config/config_component_e3sm.xml
Comment (darincomeau) [reply driver-mct/cime_config/config_component_e3sm.xml:182]: yes, will do

**Comment by jonbob** [comment]
NML and regular DIFFs blessed on pm-cpu

**Pull request merged**
